src/VecSim/algorithms/hnsw/hnsw_wrapper.cpp

```diff
@@ -81,6 +81,17 @@ VecSimResolveCode HNSWIndex::resolveParams(VecSimRawParam *rparams, int paramNum
     return (VecSimResolveCode)VecSim_OK;
 }
 
+double HNSWIndex::getDistanceFrom(size_t label, const void *vector_data) {
+    if (this->metric == VecSimMetric_Cosine) {
+        // TODO: need more generic
+        float normalized_data[this->dim];
+        memcpy(normalized_data, vector_data, this->dim * sizeof(float));
+        float_vector_normalize(normalized_data, this->dim);
+        return this->hnsw->getDistanceByLabelFromPoint(label, normalized_data);
```

Comment:
please be consistent with the implementation of FLAT and HNSW

Reply:
what do you mean? to have the normalized vector outside the `if`?